zomes/trust_atom/src/lib.rs

```diff
@@ -33,15 +33,17 @@ pub fn delete_trust_atoms(target: AnyLinkableHash) -> ExternResult<DeleteReport>
   let agent_pubkey = agent_info()?.agent_initial_pubkey;
 
   // Forward Links
-  let forward_links = get_links(agent_pubkey.clone(), LinkTypes::TrustAtom, None)?;
+  let forward_links =
+    get_links(GetLinksInputBuilder::try_new(agent_pubkey.clone(), LinkTypes::TrustAtom)?.build())?;
```

Comment:
https://docs.rs/hdk/0.3.1/hdk/link/builder/struct.GetLinksInputBuilder.html defaults GetOptions and sets other options to None:
`pub struct GetLinksInput {
    pub base_address: [HoloHash](https://docs.rs/hdk/0.3.1/hdk/prelude/struct.HoloHash.html)<[AnyLinkable](https://docs.rs/hdk/0.3.1/hdk/prelude/hash_type/enum.AnyLinkable.html)>,
    pub link_type: [LinkTypeFilter](https://docs.rs/hdk/0.3.1/hdk/prelude/enum.LinkTypeFilter.html),
    pub get_options: [GetOptions](https://docs.rs/hdk/0.3.1/hdk/prelude/struct.GetOptions.html),
    pub tag_prefix: [Option](https://doc.rust-lang.org/nightly/core/option/enum.Option.html)<[LinkTag](https://docs.rs/hdk/0.3.1/hdk/prelude/struct.LinkTag.html)>,
    pub after: [Option](https://doc.rust-lang.org/nightly/core/option/enum.Option.html)<[Timestamp](https://docs.rs/hdk/0.3.1/hdk/prelude/struct.Timestamp.html)>,
    pub before: [Option](https://doc.rust-lang.org/nightly/core/option/enum.Option.html)<[Timestamp](https://docs.rs/hdk/0.3.1/hdk/prelude/struct.Timestamp.html)>,
    pub author: [Option](https://doc.rust-lang.org/nightly/core/option/enum.Option.html)<[HoloHash](https://docs.rs/hdk/0.3.1/hdk/prelude/struct.HoloHash.html)<[Agent](https://docs.rs/hdk/0.3.1/hdk/prelude/hash_type/struct.Agent.html)>>,
}`
so for a utility function we will still need to handle optional parameters if using the builder, however a different potential change could be to leave the GetLinksInput struct in the pass through, for example:

`get_links(
      GetLinksInput {
           pub base_address: agent_pubkey,
           pub link_type: LinkTypes::TrustAtom,
           pub get_options: GetOptions::default(),
           pub tag_prefix: None,    
           pub after: None,
           pub before: None,
           pub author: None,
      }`
while not necessary cleaner, it would be more explicit

Reply:
> thus for a utility function we would still need to handle optional parameters if using the builder

not necessarily, if all of our usages of the function follow the same pattern and we don’t have to allow options in the utility function signature.

Reply:
I'm thinking that in the future not all of the usages would be the same if we decided to leverage the new query parameters, I could especially see the 'before & after' coming in handy for filtering by timestamp

perhaps for now I could go ahead and implement the utility function and call it something like get_links_default_options to distinguish

or proceed as recommended and maybe just make a brief comment for it

Reply:
I agree it would make things a bit more cleaner for now, however I lean toward keeping the builder pattern for future flexibility, since it makes it convenient to chain additional parameters if we ever do decide to add query parameters, as well as it is recommended in the migration guide.  I think we may end up wanting to leverage the new ways to filter a get_links query at some point and seems like we might use it eventually anyway. (I could see the 'before' & 'after' options coming in handy for filtering timestamps [eg. MewsFeed use-case] )

link_tag is already an optional parameter and the some case is used here:

https://github.com/trustgraph/trustgraph-holochain/blob/57b4089ee1572b02d958e82f9ba46f699cb10259/zomes/trust_atom/src/trust_atom.rs#L275


If I were to proceed as recommended, I think it might make sense to leave the builder out of the util function and just pass through the variables into the GetLinksInput struct as seen above with the given defaults, so as not to have to match on the link_tag option

otherwise the util function would like like this:

```
pub fn get_links(
  base: AnyLinkableHash,
  link_type: LinkTypes,
  link_tag: Option<LinkTag>,
) -> ExternResult<Vec<Link>> {
  if let Some(link_tag) = link_tag {
    let links = hdk::link::get_links(
      GetLinksInputBuilder::try_new(base, link_type)?
        .tag_prefix(link_tag)
        .build(),
    )?;
    Ok(links)
  } else {
    let links = hdk::link::get_links(GetLinksInputBuilder::try_new(base, link_type)?.build())?;
    Ok(links)
  }
}
```
which isn't bad, but could get lengthy if the signature changes to add more options